src/App.tsx

```diff
@@ -1,39 +1,66 @@
 import React from 'react';
+import { Clock } from './components/Clock';
 import './App.scss';
 
+const NAME_CHANGE_DELAY = 3300;
+
+type State = {
+  clockName: string;
+  hasClock: boolean;
+};
+
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+export class App extends React.PureComponent<{}, State> {
+  state: State = {
+    hasClock: true,
+    clockName: 'Clock-0',
```

Comment:
For the initialized component you can `getRandomName` too

Reply:
Sorry, I didn't get it. 'Clock-0' must be a default value to the App state.